### PR TITLE
Update devices model

### PR DESCRIPTION
**Pull request by hnico**

### CODE REVIEW

File: mds/enums.py [hunk 1]
```diff
@@ -8,3 +8,12 @@
     ("unavailable", _("Unavailable")),
     ("removed", _("Removed")),
 ]
+DEVICE_TYPE_CHOICES = [
+    ("bike", _("Bike")),
```
Comment (sebdiem) [inline mds/enums.py]: You need to manually execute `python manage.py makemessages` to update the translation files.

File: mds/models.py [hunk 2]
```diff
@@ -44,6 +45,11 @@ class Device(ProviderModel):
     model = UnboundedCharField(default=str)
     status = UnboundedCharField(choices=enums.DEVICE_STATUS_CHOICES)
     point = gis_models.PointField(null=True)
+    type = UnboundedCharField(choices=enums.DEVICE_TYPE_CHOICES)
```
Comment (sebdiem) [inline mds/models.py]: category instead of type ? this avoids shadowing a python standard function.

File: mds/locale/fr_FR/LC_MESSAGES/django.po [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+msgstr ""
+
+msgid "Car"
+msgstr ""
+
+msgid "Electric"
+msgstr ""
+
+msgid "Combustion"
+msgstr ""
```
Comment (lionel-panhaleux) [inline mds/locale/fr_FR/LC_MESSAGES/django.po]: we need fr_FR translations to be set manually.

File: mds/views.py [hunk 4]
```diff
@@ -116,7 +117,14 @@ class DeviceViewSet(
     UpdateOnlyModelMixin,
     MultiSerializerViewSet,
 ):
-    queryset = models.Device.objects.all()
+    queryset = models.Device.objects.all().prefetch_related(
+        Prefetch(
+            "telemetries",
+            queryset=models.Telemetry.objects.order_by("-timestamp"),
+            to_attr="ordered_telemetries",
+        )
```
Comment (lionel-panhaleux) [inline mds/views.py]: This is better as you do not get the N+1 problem anymore, still you fetch all telemetries for each vehicle, which seems a lot just to get the last telemetry date. Subquery might do a better jog than prefetch, cf. https://docs.djangoproject.com/en/2.1/ref/models/expressions/#subquery-expressions

File: mds/locale/fr_FR/LC_MESSAGES/django.po (hunk 3 above)
Comment (lionel-panhaleux) [inline mds/locale/fr_FR/LC_MESSAGES/django.po]: We can put French translations manually right now ;-)

File: mds/serializers.py [hunk 5]
```diff
@@ -131,8 +131,9 @@ class Device(BaseModelSerializer):
     """
 
     id = serializers.UUIDField(help_text="Unique device identifier (UUID)")
-    provider = serializers.UUIDField(
-        help_text="A unique ID identifying the service provider (UUID)"
+    provider = serializers.CharField(
+        source="provider.name",
+        help_text="A unique ID identifying the service provider (UUID)",
```
Comment (sebdiem) [inline mds/serializers.py]: help_text not updated ;)

**Pull request merged**